fwdpy11/_functions/import_demes.py

```diff
@@ -354,7 +342,14 @@ def _get_model_times(dg: demes.Graph) -> _ModelTimes:
         mig_starts = [m.start_time for m in dg.migrations if m.start_time != math.inf]
         mig_ends = [m.end_time for m in dg.migrations if m.start_time == math.inf]
         pulse_times = [p.time for p in dg.pulses]
-        model_start_time = max(ends_inf + starts + mig_starts + mig_ends + pulse_times)
+
+        # The forward-time model with start with a generation 0,
+        # which is the earliest end point of a deme with start time
+        # of inf, minus 1.  That definition is forwards in time, so we
+        # ADD one to the backwards-in-time demes info.
+        model_start_time = (
+            max(ends_inf + starts + mig_starts + mig_ends + pulse_times) + 1
```

Comment:
Will we need to make sure, if we are doing some precapitation, that we haven't inserted an extra generation in between the two halves of the simulation?

Reply:
Yes, that is the potential "gotcha" here, I think, and it deserves some consideration.  If the pre/re-capitation is from the msprime continuous model, then there's no issue (??).  But if it is the DTWF, then we have to be careful?